plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/transactionlog/TableSnapshot.java

```diff
@@ -229,7 +235,8 @@ private Stream<DeltaLakeTransactionLogEntry> getCheckpointTransactionLogEntries(
                 metadataEntry,
                 stats,
                 parquetReaderOptions,
-                checkpointRowStatisticsWritingEnabled));
+                checkpointRowStatisticsWritingEnabled,
```

Comment:
`domainCompactionThreshold` is highly correlated with `parquetReaderOptions`.
Can you investigate whether it would make sense adding this setting to `io.trino.parquet.ParquetReaderOptions` ?

Reply:
domainCompactionThreshold is a connector level config today.
To move it to `ParquetReaderOptions`, we'd need to also add the same config to `OrcReaderOptions`. Maybe that's better, but let's keep that outside the scope of this PR.

Reply:
I see `domainCompactionThreshold` parameter passed around in a lot of sink & page source classes only to be eventually used in `io.trino.parquet.predicate.TupleDomainParquetPredicate#matches(io.trino.parquet.BloomFilterStore)`. 

Note that if there is no bloom filter used in `ParquetReaderOptions`, this parameter is meaningless.

Reply:
It's not ideal, but there isn't a straightforward way around it.
If we make that config a part of ParquetReaderOptions, then it needs to be added to OrcReaderOptions as well and removed from connector level config.
That may work for hive but for delta and iceberg, it's not clear that domain compaction should be in reader options. There we deal with predicates from the connector level metadata as well and not just from the metadata in the file formats.